File: packages/sdk/src/core/adapters/network-adapter.ts
export type NetworkState = {
  isConnected: boolean;
};

export type NetworkStateListener = (state: NetworkState) => void;
export type UnsubscribeFn = () => void;

export type NetworkAdapter = {
  fetchNetworkState(): Promise<NetworkState>;
  addNetworkListener(listener: NetworkStateListener): UnsubscribeFn;
};

File: packages/swift/Sources/PhaseAnalytics/Core/Adapters/NetworkAdapter.swift
import Foundation

public typealias NetworkStateListener = @Sendable (NetworkState) -> Void
public typealias UnsubscribeFn = @Sendable () -> Void

public protocol NetworkAdapter: Sendable {
    func fetchNetworkState() async -> NetworkState
    func addNetworkListener(_ listener: @escaping NetworkStateListener) -> UnsubscribeFn
}
